src/reducers/subnetwork.test.js

```diff
@@ -0,0 +1,148 @@
+import {
+  ADD_SUPER_NODE,
+  LOAD_NETWORK,
+  NEW_NETWORK,
+  REMOVE_SUPER_NODE,
+} from 'actions';
+
+import reducer from './subnetwork';
+
+describe('Subnetwork Reducers', () => {
+  describe('DEFAULT', () => {
+    it('returns an empty array', () => {
+      expect(reducer(undefined, {})).toEqual([]);
+    });
+  });
+
+  describe('NEW_NETWORK', () => {
+    it('returns an empty array', () => {
+      expect(reducer(undefined, {
+        type: NEW_NETWORK,
+      })).toEqual([]);
+    });
+  });
+
+  describe('LOAD_NETWORK', () => {
+    it('returns subnetworks from action payload', () => {
+      const subnetworks = [1, 2, 3];
+      const action = {
+        type: LOAD_NETWORK,
+        payload: {
+          state: {
+            network: { subnetworks },
+          },
+        },
+      };
+
+      expect(reducer(undefined, action)).toEqual(subnetworks);
+    });
+
+    it('returns empty array if action payload has no subnetwork', () => {
+      const action = {
+        type: LOAD_NETWORK,
+        payload: {
+          state: {
+            network: {},
+          },
+        },
+      };
+
+      expect(reducer(undefined, action)).toEqual([]);
+    });
+  });
+
+  describe('ADD_SUPER_NODE', () => {
+    it('returns state with added super node', () => {
+      const state = [
+        { id: 1 },
+      ];
+
+      const nodes = [1, 2, 3];
+      const positions = [1, 2, 3];
+
+      const action = {
+        type: ADD_SUPER_NODE,
+        payload: {
+          state: {
+            network: { id: 2 },
+            nodes,
+            positions,
+          },
+        },
+      };
+
+      expect(reducer(state, action)).toEqual([
+        { id: 1 },
+        {
+          id: 2, nodes, positions, color: '#FFFF00',
+        },
+      ]);
+    });
+
+    it('returns state with added network when payload has network only', () => {
+      const state = [
+        { id: 1 },
+      ];
+
+      const action = {
+        type: ADD_SUPER_NODE,
+        payload: {
+          state: {
+            network: { id: 2 },
+          },
+        },
+      };
+
+      expect(reducer(state, action)).toEqual([
+        { id: 1 },
+        { id: 2 },
+      ]);
+    });
+
+    it('returns new state when payload has state only', () => {
+      const state = [
+        { id: 1 },
+      ];
+
+      const action = {
+        type: ADD_SUPER_NODE,
+        payload: {
+          state: {},
+        },
+      };
+
+      expect(reducer(state, action)).toEqual([
+        { id: 1 }, undefined,
+      ]);
+    });
```

Comment:
Nice, but I think you can rewrite like this:
```js
describe('ADD_SUPER_NODE', () => {
  // you can declare `state` here, since all test are using it
  const state = [
    { id: 1 },
  ];

  describe('When action has a network with nodes and positions', () => {
    // Setup here and keep `it` as clear as possible 
    const nodes = [1, 2, 3];
    const positions = [1, 2, 3];

    const action = {
      type: ADD_SUPER_NODE,
      payload: {
        state: {
          network: { id: 2 },
          nodes,
          positions,
        },
      },
    };

    it('adds network, nodes and positions on state', () => {
      expect(reducer(state, action)).toEqual([
        { id: 1 },
        {
          id: 2, nodes, positions, color: '#FFFF00',
        },
      ]);
    });
  })
  
  describe('When action has a network without nodes and positions', () => {
    const action = {
      type: ADD_SUPER_NODE,
      payload: {
        state: {
          network: { id: 2 },
        },
      },
    };

    it('adss network on state', () => {
      expect(reducer(state, action)).toEqual([
        { id: 1 },
        { id: 2 },
      ]);
    });
  })

  describe('When action has not network', () => {
    const action = {
      type: ADD_SUPER_NODE,
      payload: {
        state: {},
      },
    };

    it('adds undefined on state', () => {
      expect(reducer(state, action)).toEqual([
        { id: 1 }, 
        undefined,
      ]);
    });
  })
});
```